app/services/mark_response_as_sent_service.rb

```diff
@@ -0,0 +1,45 @@
+class MarkResponseAsSentService
+
+  attr_reader :result
+
+  def initialize(kase, user, params)
+    @kase = kase
+    @params = params
+    @user = user
+    @result = nil
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      @kase.prepare_for_respond
+      @kase.ico? ? respond_to_ico : respond_to_non_ico
```

Comment:
Aren't SAR cases like ICOs here, i.e. they get responded + closed in a single action?

Reply:
And ... does this really need to be separated out in the functions below?

Reply:
ICO cases I think are different, they go approved (awaiting_dispatch) to responded (awaiting_ico_decision), to closed

Reply:
No sure about merging the methods, it makes sense but having tried it out, it is quite tricky to head you head around